index.js

```diff
@@ -357,153 +394,167 @@ const sortExports = onObject((exports) => {
-  { key: 'npmPackageJsonLintConfig', over: sortObject },
-  { key: 'npmpackagejsonlint', over: sortObject },
-  { key: 'release', over: sortObject },
-  { key: 'remarkConfig', over: sortObject },
-  { key: 'stylelint' },
-  { key: 'ava', over: sortObject },
-  { key: 'jest', over: sortObject },
-  { key: 'jest-junit', over: sortObject },
-  { key: 'jest-stare', over: sortObject },
-  { key: 'mocha', over: sortObject },
-  { key: 'nyc', over: sortObject },
-  { key: 'c8', over: sortObject },
-  { key: 'tap', over: sortObject },
-  { key: 'oclif', over: sortObjectBy(undefined, true) },
-  { key: 'resolutions', over: sortObject },
-  { key: 'overrides', over: sortDependenciesLikeNpm },
-  { key: 'dependencies', over: sortDependenciesLikeNpm },
-  { key: 'devDependencies', over: sortDependenciesLikeNpm },
-  { key: 'dependenciesMeta', over: sortObjectBy(sortObjectByIdent, true) },
-  { key: 'peerDependencies', over: sortDependenciesLikeNpm },
-  // TODO: only sort depth = 2
-  { key: 'peerDependenciesMeta', over: sortObjectBy(undefined, true) },
-  { key: 'optionalDependencies', over: sortDependenciesLikeNpm },
-  { key: 'bundledDependencies', over: uniqAndSortArray },
-  { key: 'bundleDependencies', over: uniqAndSortArray },
-  /* vscode */ { key: 'extensionPack', over: uniqAndSortArray },
-  /* vscode */ { key: 'extensionDependencies', over: uniqAndSortArray },
-  { key: 'flat' },
-  { key: 'packageManager' },
-  { key: 'engines', over: sortObject },
-  { key: 'engineStrict', over: sortObject },
-  { key: 'volta', over: sortVolta },
-  { key: 'languageName' },
-  { key: 'os' },
-  { key: 'cpu' },
-  { key: 'preferGlobal', over: sortObject },
-  { key: 'publishConfig', over: sortObject },
-  /* vscode */ { key: 'icon' },
-  /* vscode */ {
-    key: 'badges',
-    over: onArray((badge) => badge.map(sortVSCodeBadgeObject)),
-  },
-  /* vscode */ { key: 'galleryBanner', over: sortObject },
-  /* vscode */ { key: 'preview' },
-  /* vscode */ { key: 'markdown' },
-  { key: 'pnpm', over: sortPnpmConfig },
-]
+/**
+ * Creates fields configuration with appropriate sorting function for
+ * dependencies based on the package manager
+ *
+ * fields marked `vscode` are for `Visual Studio Code extension manifest` only
+ * https://code.visualstudio.com/api/references/extension-manifest
+ * Supported fields:
+ * publisher, displayName, categories, galleryBanner, preview, contributes,
+ * activationEvents, badges, markdown, qna, extensionPack,
+ * extensionDependencies, icon
+ *
+ * field.key{string}: field name
+ * field.over{function}: sort field subKey
+ *
+ * @param {'npm'|'yarn'|'pnpm'} packageManager - The package manager to use
+ * @returns {Array} fields configuration array
+ */
+function createFields(packageManager) {
+  // npm uses locale-aware comparison, yarn and pnpm use simple string comparison
+  const sortDependencies =
+    packageManager === 'npm'
+      ? sortObjectAlphabeticallyWithLocale
+      : sortObjectAlphabetically
```

Comment:
This might be simpler if you created a `sortObjectAlphabeticallyDetectingPackageManager` or something - making the detection just-in-time, rather than doing so ahead of time. I could imagine it being a smaller diff. I'm not precious about either approach though, so this LGTM.